src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/FilingsService.java

```diff
@@ -209,6 +210,11 @@ private void collectUpdateSubmissionData(UpdateSubmission updateSubmission,
       updateSubmission.getChanges()
               .addAll(overseasEntityChangeService.collateOverseasEntityChanges(publicPrivateOeData,
                       submissionDto, logMap));
+
+      var existingCompanyName = new EntityNameDto();
+        existingCompanyName.setName(publicPrivateOeData.getLeft().getCompanyName());
+        updateSubmission.getUserSubmission().setEntityName(existingCompanyName);
```

Comment:
Does this not overwrite the entity name in the user submission? What if the user is requesting a name change? This  will reset the changed name to the old name from the API.
Suggest a new filed in the updateSubmission (called entityName) which we set to be publicPrivateOeData.getLeft().getCompanyName().
This new field will have to be picked up on the XML CHIPS consumer side.

Reply:
I think we are doing this after the name change calculation -- would that be ok ?
@azafarCH -- check if we are doing this after ....

Reply:
Would still not be correct to change the user submission which we may pass in XML or log. Suggested adding a new (JSON) field to the UpdateSubmssion Dto and just setting that to the name from public api. New field needs to be picked up by CHIPS filing consumer and put in XML.

Reply:
We've made changes, added the new field and mapped @AndreKramer7654